forte/pipeline.py

```diff
@@ -281,24 +319,63 @@ def _dump_to_config(self):
         Returns:
             dict: A dictionary storing IR.
         """
-        configs: List[Dict] = []
-        configs.append(
+        configs: Dict = {
+            "forte_ir_version": self.FORTE_IR_VERSION,
+            "components": list(),
+            "states": dict(),
+        }
+
+        # Serialize pipeline components
+        configs["components"].append(
             {
                 "type": ".".join(
                     [self._reader.__module__, type(self._reader).__name__]
                 ),
                 "configs": self._reader_config.todict(),
             }
         )
-        for component, config in zip(self.components, self.component_configs):
-            configs.append(
+        for component, config, selector in zip(
+            self.components, self.component_configs, self._selectors
+        ):
+            configs["components"].append(
                 {
                     "type": ".".join(
                         [component.__module__, type(component).__name__]
                     ),
                     "configs": config.todict(),
+                    "selector": {
+                        "type": ".".join(
+                            [selector.__module__, type(selector).__name__]
+                        ),
+                        # TODO: This presumes that class attributes' names are
+                        # the same as the paramaters' names passed to
+                        # selector's constructor, which may not be always true.
+                        "kwargs": selector.__dict__ or None,
```

Comment:
Maybe replace `None` with `{}` since you will use`{}` in place of `None` any ways?